src/components/AnnotationViewer.vue

```diff
@@ -405,29 +366,54 @@ export default class AnnotationViewer extends Vue {
     this.textLayer.draw();
   }
 
-  shouldDisplayAnnotationWithChannel(channelId: number): boolean {
-    return this.visibleChannels.includes(channelId);
+  get validLayers() {
+    return this.layers.slice(
+      this.lowestLayer,
+      this.lowestLayer + this.layerCount
+    );
   }
 
-  shouldDisplayAnnotationWithLocation(location: IAnnotationLocation): boolean {
+  // Should the annotation be displayed in the layer
+  // The precheck checks if the annotation is displayable and the layer valid
+  layerShouldDisplayAnnotation(
+    layer: IDisplayLayer,
+    annotation: IAnnotation,
+    precheck: boolean = true
+  ) {
+    if (precheck) {
+      if (
+        !this.validLayers.some(validLayer => validLayer.id === layer.id) ||
+        !this.displayableAnnotations.some(
+          displayableAnnotation => displayableAnnotation.id === annotation.id
+        )
```

Comment:
`shouldDisplayAnnotation` calls `layerShouldDisplayAnnotation`
(`layerShouldDisplayAnnotation` is the lowest level function)

Reply:
This is an optimization: when `shouldDisplayAnnotation` calls `layerShouldDisplayAnnotation` it gives the argument `precheck = false` so that the checks for layer validity and annotation validity are not made.
And it does so because it knows that the layer belongs to `validLayers` and that the check for the annotation validity has been done once for all the calls to `layerShouldDisplayAnnotation`.